w2w/w2w.py

```diff
@@ -246,21 +246,36 @@ def wrf_remove_urban(
                 newgreenf[:,i,j]=auxg
 
             if luf.isel(south_north=i,west_east=j,land_cat=12)>0.:
-                dis = calc_distance_coord(
-                                         lat.where(
-                                                    (luf.isel(land_cat=12)==0.) &
-                                                    (luf.isel(land_cat=16)==0.) &
-                                                    (luf.isel(land_cat=20)==0.)
-                                                  ),
-
-                                         lon.where(
-                                                    (luf.isel(land_cat=12)==0.) &
-                                                    (luf.isel(land_cat=16)==0.) &
-                                                    (luf.isel(land_cat=20)==0.)
-                                                  ),
-                                         lat.isel(south_north=i,west_east=j),
-                                         lon.isel(south_north=i,west_east=j)
-                                         )
+                if len(luf.land_cat)>=20: #USING MODIS_LAKE
```

Comment:
Do I understand this correctly that datasets without lakes have only landuse categories up to 19? So all other categories are not present? So it is impossible that classes 1 - 19 are there, but 20 and 21 are missing and then 22 to 40 are there again?
```console
(Pdb) p luf.land_cat
<xarray.DataArray 'land_cat' (land_cat: 41)>
array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17,
       18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35,
       36, 37, 38, 39, 40])
Dimensions without coordinates: land_cat
(Pdb) p len(luf.land_cat)
41
```
So our example dataset is `MODIS_LAKE` ?

Reply:
The variable land use fraction (luf) is a three dimensional array (lat, lon, land_cat; not necessarily in this order). It specifies the fraction of each category in each grid-cell. So if a grid point has 50% urban (cat 13) and 50% croplands (cat 12), luf will take 0.5 in levels 12th and 13th. Because python is 0-indexed, it will actually take those values in levels 11th and 12th. 

The size of luf is created depending on the available categories, but always keeping correspondence with the numbers assigned to each category. So if the input data has info for category 30 (whatever it be), it create a luf with 30 levels, regardless of whether there is info for levels below 30. 

So the short answer is you understood well. Our example is MODIS_LAKE. You can have categories up to 20 only, but you can't have classes 1-20, then 21 missing, then again from 22 to 40. In that latest case, class 21 will simply be filled with all zeroes.

Not sure I explained my self very well.

Reply:
Thank you for the clarification and good explanation. I was just wondering if it should have been `in` instead of `len >=` but since all categories below the highest one are created anyways, this is fine. Thank you!